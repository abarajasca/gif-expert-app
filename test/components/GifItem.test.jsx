import { render, screen } from "@testing-library/react"
import GifItem from "../../src/components/GifItem"

describe('GifItem test', () => {
  const title = "The Title"
  const url   = 'https://www.google.com/image'

    test('should match with snapshot', () => { 
        const { container } = render(<GifItem title={ title } url={ url} />)    
        expect( container ).toMatchSnapshot()    
    })

    test('should show image with url and alt defined', () => {
        render(<GifItem title={ title } url={ url } />)
        const img = screen.getByRole('img')    
        expect( img.alt ).toBe( title )
        expect( img.src ).toBe( url )
    })
   
})
