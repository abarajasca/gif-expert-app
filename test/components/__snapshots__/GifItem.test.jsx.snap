// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem test should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="The Title"
      src="https://www.google.com/image"
    />
    <p>
      The Title
    </p>
  </div>
</div>
`;
